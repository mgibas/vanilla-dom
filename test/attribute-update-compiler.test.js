const compiler = require('../src/attribute-update-compiler')

describe('attribute-update-compiler', () => {
  describe('on compile', () => {
    it('single path', () => {
      expect(compiler.compile('testElement', 'foo',{
        paths: ['bar'],
        template: () => '"test-template-string"'
      })).toMatchSnapshot();
    })
    it('multiple paths', () => {
      expect(compiler.compile('testElement', 'foo',{
        paths: ['bar', 'foo'],
        template: () => '"test-template-string"'
      })).toMatchSnapshot();
    })
  })
})
