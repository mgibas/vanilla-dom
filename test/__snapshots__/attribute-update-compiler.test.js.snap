// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attribute-update-compiler on compile class attribute 1`] = `
"_updaters.push({
  update: st => {
    let val = 1;
    if (val === null || val === undefined)
      val = 'awesome';
    let currentVal = testElement.getAttribute('class');
    if (typeof val === 'boolean' && val) {
      if (currentVal !== '')
        testElement.setAttribute('class', '');
      return;
    }
    if (val && typeof val === 'string' || typeof val === 'number') {
      if (currentVal != val) {
        testElement.className = val;
      }
      return;
    }
    testElement.removeAttribute('class');
  }
});"
`;

exports[`attribute-update-compiler on compile multiple paths 1`] = `
"_updaters.push({
  update: st => {
    let val = 1;
    if (val === null || val === undefined)
      val = 'test-template-string';
    let currentVal = testElement.getAttribute('foo');
    if (typeof val === 'boolean' && val) {
      if (currentVal !== '')
        testElement.setAttribute('foo', '');
      return;
    }
    if (val && typeof val === 'string' || typeof val === 'number') {
      if (currentVal != val) {
        testElement.setAttribute('foo', val);
      }
      return;
    }
    testElement.removeAttribute('foo');
  }
});"
`;

exports[`attribute-update-compiler on compile single path 1`] = `
"_updaters.push({
  update: st => {
    let val = 1;
    if (val === null || val === undefined)
      val = 'test-template-string';
    let currentVal = testElement.getAttribute('foo');
    if (typeof val === 'boolean' && val) {
      if (currentVal !== '')
        testElement.setAttribute('foo', '');
      return;
    }
    if (val && typeof val === 'string' || typeof val === 'number') {
      if (currentVal != val) {
        testElement.setAttribute('foo', val);
      }
      return;
    }
    testElement.removeAttribute('foo');
  }
});"
`;
